### ISSUE TITLE
remove `invalid-type-guard-call` diagnostic

### ISSUE THREAD

**Issue by DetachHead** [astral-sh/ty#3991]
> i'm still a bit confused about this rule. the docs stilll say this:
> 
> Starred (`is_str(*a)`), literal (`is_str(42)`) and other non-symbol-like expressions are invalid as narrowing targets.
> 
> but that doesn't seem to actually be the case
> 
> i also don't understand the new example:
> 
> ```py
> is_int(value=1)
> ```
> 
> specifying the narrowing target with a keyword arg *does* report the error, regardless of whether the value is a literal expression or not:
> 
> ```py
> is_int(value=value)  # Error
> ```
> 
> i also don't get why this is even an error at all. ty should still be able to tell that this value is a narrowing target. [pyright is fine with it](https://basedpyright.com/?pythonVersion=3.15&typeCheckingMode=all&reportUninitializedInstanceVariable=false&code=GYJw9gtgBALgngBwJYDsDmUkQWEMoAqiApgJIDOANFCMQG7ECGANgPrwLEBQXAJscEzlWqGAAo6LAK7EAXFDAAjAFbEAxjACUUALQA%2BQiQoBtUQF1ZXKNZrEYUkCiGpyMRijXEJ04tVGaePgEoYBRvZhl5JVUNTUsbTEEkYVFwmQBeSQjiOOsAYigAURBwEHkURhKwAHdUDDcQNDsoBEZycmJeKEU4KABrYjhq3F4rBJtaBhZ2EjScoA) 

 _Originally posted by @DetachHead in [#3735](https://github.com/astral-sh/ty/issues/3735#issuecomment-4698300648)_

https://docs.astral.sh/ty/reference/rules/#invalid-type-guard-call

also, with this example:

```py
def is_int(value: object = object()) -> TypeIs[int]:
    
return isinstance(value, int)


# no positional narrowing target
is_int()  # error
```

if it's invalid to call a `TypeIs` without an argument, i think there should be an error on the declaration instead of the call, because it means it's the function signature that's incorrect.

my main point though is that i don't see why a `TypeIs`/`TypeGuard` call that doesn't have a narrowing target should even be an error in the first place. some examples that should help illustrate my point:
```py
if isinstance(foo(), int): # should this be an error because there's no narrowing target?
    ...

if foo() is 1: # should this be an error because there's no narrowing target?
    ...
```

**Comment by carljm** [comment]
I agree with this; I can't find any convincing rationale for `invalid-type-guard-call` diagnostic to exist at all. It dates all the way back to https://github.com/astral-sh/ruff/pull/18589, the PR that first introduced `TypeIs` support, but no rationale for it is presented there, and a comment in the code even asks whether it should be an error. There's no indication anywhere in the conformance suite or spec that calls to a type-guard function which don't have a narrowing target should error, and I don't think it's useful; it just needlessly limits the runtime use of those functions.

I think `invalid-type-guard-call` should be removed, and calls to type-guard functions which don't have a narrowing target should just not narrow anything. And we should add support for narrowing keyword arguments, e.g. `is_int(value=x)`, when they specify what would otherwise be the first positional parameter. Mypy and pyright do this, and IMO it makes sense to do it; `is_int(value=x)` should be equivalent to `is_int(x)`.
